### ISSUE TITLE
ETSModel cannot predict in sample with multiplicative error, trend, or seasonal

### ISSUE THREAD

**Issue by albertmichaelj** [statsmodels/statsmodels#7120]
#### Describe the bug

If you try to predict in sample using an ETS model with `error = "mul"`, `trend = "mul"`, or `seasonal = "mul"`, you get an error `TypeError: Series.name must be a hashable type`.

#### Code Sample, a copy-pastable example if possible
This example is adapted from the example in the docs [here](https://www.statsmodels.org/dev/examples/notebooks/generated/ets.html).

```python
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
%matplotlib inline
from statsmodels.tsa.exponential_smoothing.ets import ETSModel

plt.rcParams['figure.figsize'] = (12, 8)

austourists_data = [
    30.05251300, 19.14849600, 25.31769200, 27.59143700,
    32.07645600, 23.48796100, 28.47594000, 35.12375300,
    36.83848500, 25.00701700, 30.72223000, 28.69375900,
    36.64098600, 23.82460900, 29.31168300, 31.77030900,
    35.17787700, 19.77524400, 29.60175000, 34.53884200,
    41.27359900, 26.65586200, 28.27985900, 35.19115300,
    42.20566386, 24.64917133, 32.66733514, 37.25735401,
    45.24246027, 29.35048127, 36.34420728, 41.78208136,
    49.27659843, 31.27540139, 37.85062549, 38.83704413,
    51.23690034, 31.83855162, 41.32342126, 42.79900337,
    55.70835836, 33.40714492, 42.31663797, 45.15712257,
    59.57607996, 34.83733016, 44.84168072, 46.97124960,
    60.01903094, 38.37117851, 46.97586413, 50.73379646,
    61.64687319, 39.29956937, 52.67120908, 54.33231689,
    66.83435838, 40.87118847, 51.82853579, 57.49190993,
    65.25146985, 43.06120822, 54.76075713, 59.83447494,
    73.25702747, 47.69662373, 61.09776802, 66.05576122,
]
index = pd.date_range("1999-03-01", "2015-12-01", freq="3MS")
austourists = pd.Series(austourists_data, index=index, name='test')
austourists.plot()
plt.ylabel('Australian Tourists');

# fit in statsmodels
model = ETSModel(austourists, error="mul", trend="add", seasonal="add",
                damped_trend=True, seasonal_periods=4)
fit = model.fit()

# fit with R params
params_R = [
    0.35445427, 0.03200749, 0.39993387, 0.97999997, 24.01278357,
    0.97770147, 1.76951063, -0.50735902, -6.61171798, 5.34956637
]
fit_R = model.smooth(params_R)

austourists.plot(label='data')
plt.ylabel('Australian Tourists')

fit.fittedvalues.plot(label='statsmodels fit')
fit_R.fittedvalues.plot(label='R fit', linestyle='--')
plt.legend();

pred = fit.get_prediction(start='2014', end='2020')
```
<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

I have searched for this bug, but I cannot find an open issue.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `master`. If your problem has been fixed in an unreleased version, you might be able to use `master` until a new release occurs. 

I'm using `0.12.0` installed using conda.

**Note**: If you are using a released version, have you verified that the bug exists in the master branch of this repository? It helps the limited resources if we know problems exist in the current master so that they do not need to check whether the code sample produces a bug in the next release.

I have not verified that this bug exists in master.

</details>

#### Expected Output

I expect that you should be able to predict in sample exactly like when using additive errors, trends, and seasonalities.

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
Python: 3.7.7.final.0
OS: Darwin 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

statsmodels
===========

Installed: 0.12.0 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.21 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/Cython)
numpy: 1.19.2 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/numpy)
scipy: 1.5.0 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/scipy)
pandas: 1.0.5 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/pandas)
    dateutil: 2.8.1 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/dateutil)
patsy: 0.5.1 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.3.2 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/matplotlib)
    backend: module://ipykernel.pylab.backend_inline 
cvxopt: Not installed
joblib: 0.17.0 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/joblib)

Developer Tools
================

IPython: 7.18.1 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/IPython)
    jinja2: 2.11.2 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/jinja2)
sphinx: Not installed
    pygments: 2.7.1 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/pygments)
pytest: 5.4.3 (/Users/Michael/.brew/Caskroom/miniconda/base/envs/darden/lib/python3.7/site-packages/pytest)
virtualenv: Not installed

</details>

**Comment by ChadFulton** [comment]
Thanks very much for this report. Fortunately, this bug has already been fixed in master (it looks like by #7055). We will have a v0.12.1 release at the end of this week with the accumulated bug fixes, including this one.

Closing since already fixed, but feel free to follow up.

**Comment by albertmichaelj** [comment]
Thanks for the response! And great to know it has been fixed in master. Looking forward to the bug fix!
